console.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/miquella/vaulted/lib"
+	"github.com/pkg/browser"
+)
+
+var (
+	ErrInvalidTemporaryCredentials = errors.New("Temporary session credentials found. Console cannot be opened with temporary credentials.\nIf the AWS Key is set in a vault, the permanent credentials can be used by specifying the vault name.")
+	ErrInvalidDuration             = errors.New("Console duration must be between 15m and 12h.")
+	ErrNoCredentialsFound          = errors.New("No credentials found. Console cannot be opened.")
+)
+
+const (
+	CONSOLE_URL = "https://console.aws.amazon.com/console/home"
+	SIGNIN_URL  = "https://signin.aws.amazon.com/federation"
+
+	MinConsoleDuration = 15 * time.Minute
+	MaxConsoleDuration = 12 * time.Hour
+)
+
+type Console struct {
+	VaultName string
+	Role      string
+	Duration  time.Duration
+}
+
+func (c *Console) Run(store vaulted.Store) error {
+	signinToken, err := c.getSigninToken(store)
+	if err != nil {
+		return err
+	}
+
+	// console signin
+	signinUrl, _ := url.Parse(SIGNIN_URL)
+	loginQuery := make(url.Values)
+	loginQuery.Set("Action", "login")
+	loginQuery.Set("SigninToken", signinToken)
+	loginQuery.Set("Destination", CONSOLE_URL)
+
+	signinUrl.RawQuery = loginQuery.Encode()
+	err = browser.OpenURL(signinUrl.String())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Console) getSigninToken(store vaulted.Store) (string, error) {
+	// Setup default values (may be overwritten by values from vault)
+	duration := 1 * time.Hour
+	var awsKey vaulted.AWSKey
+
+	// Override defaults with values from specified vault
+	if c.VaultName != "" {
+		v, _, err := store.OpenVault(c.VaultName)
+		if err != nil {
+			return "", err
+		}
+
+		duration = v.Duration
+
+		if v.AWSKey.Valid() {
+			awsKey = *v.AWSKey
+			if c.Role != "" {
+				awsKey.Role = c.Role
+			}
+		}
+	}
+
+	// If duration was provided through the command line overwrite with that
```

Comment:
Also, `15*time.Minute` and `12*time.Hour` should be used as consts when we do these checks.

Reply:
done